bin/clover/src/pipeline-steps/assetSpecificOverrides.ts

```diff
@@ -25,31 +28,49 @@ type OverrideFn = (spec: ExpandedPkgSpec) => void;
 
 const overrides = new Map<string, OverrideFn>([
   ["AWS::EC2::Route", (spec: ExpandedPkgSpec) => {
-    addGatewayIdSocketToEC2Route(spec);
+    // Add GatewayId Socket
+    const variant = spec.schemas[0].variants[0];
+
+    const prop = variant.domain.entries.find((p) => p.name === "GatewayId");
+
+    if (!prop) return;
+    const socket = createInputSocketFromProp(prop, "one");
+
+    setAnnotationOnSocket(socket, { tokens: ["InternetGatewayId"] });
+    setAnnotationOnSocket(socket, { tokens: ["VPNGatewayId"] });
+
+    variant.sockets.push(socket);
   }],
-]);
+  ["AWS::CloudTrail::Trail", (spec: ExpandedPkgSpec) => {
+    const variant = spec.schemas[0].variants[0];
 
-function addGatewayIdSocketToEC2Route(spec: ExpandedPkgSpec) {
-  const schema = spec.schemas[0];
-  const variant = spec.schemas[0].variants[0];
-  const domain = variant.domain;
+    for (const prop of variant.domain.entries) {
+      switch (prop.name) {
+        case "S3BucketName": {
+          const socket = createInputSocketFromProp(prop, "one", [{
```

Comment:
If you call getOrCreateInputSocketFromPop, it will actually take care of the push as well (and prevent dups from appearing).